### PR TITLE
Add Time Machine widget

### PR DESCRIPTION
**Pull request by bochoven**

### CODE REVIEW

**Comment by bochoven** [comment]
Not sure if this is ok, I copied some stuff from the CrashPlan widget. There's no script installer yet, right?

File: mr_module/TimeMachine/Providers/TimeMachineServiceProvider.php [hunk 1]
```diff
@@ -3,15 +3,24 @@
...
+            ->middleware('api')
+            ->namespace($this->namespace)
+            ->group(base_path('mr_module/TimeMachine/routes.php'));
+    }
+
     public function boot(ModuleManager $moduleManager) {
+        $this->mapApiRoutes();
         $this->loadMigrationsFrom(__DIR__.'/../migrations');
 
-        $moduleManager->add('timemachine', dirname(__DIR__))
```
Comment (mosen) [inline mr_module/TimeMachine/Providers/TimeMachineServiceProvider.php:13]: The moduleManager stuff is still needed for install/uninstall script to work.

File: mr_module/TimeMachine/TimeMachineController.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<?php
+namespace MrModule\TimeMachine;
+
+use Mr\Http\Controllers\Controller;
+use MrModule\TimeMachine\TimeMachine;
+
+class TimeMachineController extends Controller
+{
+    public function index() {
+        $results = TimeMachine::all();
```
Comment (mosen) [inline mr_module/TimeMachine/TimeMachineController.php:10]: I know i use ::all() for a lot of index() methods. In future i will have to change them to support paging, filter, and sorting.

File: mr_module/TimeMachine/TimeMachineController.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public function stats() {
+        $lastDay = new \DateTime();
+        $lastDay->sub(new \DateInterval('P1D'));
+
+        $lastWeek = new \DateTime();
+        $lastWeek->sub(new \DateInterval('P1W'));
+
+        $todayCount = TimeMachine::where('last_success', '>', $lastDay)->count();
```
Comment (mosen) [inline mr_module/TimeMachine/TimeMachineController.php:33]: This is how i am doing stats at the moment, but we CAN use a CASE() statement if it gets too slow.

**Review by mosen — APPROVED** [review]
Added some notes to explain what I'm doing.

**Pull request merged**